### PR TITLE
Improved single-crystal peak finding algorithm

### LINKED ISSUE #35925: Improved algorithm for single-crystal Bragg peak finding
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Report to:** Matthias Gutmann and Silvia Capelli (SXD)

**Description of work**
New algorithm to find single-crystal Bragg peaks in instruments with rectangular detectors (see algorithm docs for details).

**Purpose of work**
SXD need a better peak finding algorithm as they want to integrate found peaks.

**To test:**

(1) Run this script
```
ws = Load(Filename="SXD23767.raw", OutputWorkspace="SXD23767")
FindSXPeaksConvolve(InputWorkspace=ws, PeaksWorkspace='peaks_out', GetNBinsFromBackToBackParams=False, 
                    ThresholdIoverSigma=5.0)
```
(2) Try specifying `GetNBinsFromBackToBackParams=False`
```
FindSXPeaksConvolve(InputWorkspace=ws, PeaksWorkspace='peaks_out_nbins', GetNBinsFromBackToBackParams=False, 
                    NBins=7, RemoveOnEdge=True)
```
(3) Try specifying `RemoveOnEdge=True` - overlap `peaks_out` and `peaks_out_noedge` on `ws` in instrument view and check the peaks on the edge are missing (edge region is determined from kernel size).
```
FindSXPeaksConvolve(InputWorkspace=ws, PeaksWorkspace='peaks_out_noedge', GetNBinsFromBackToBackParams=True,
                    RemoveOnEdge=True)
```
Fixes #35925

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
Sorry its taken so long for me to review this! Looks good in terms of code - I think just a few typos to correct and then I'll approve

File: docs/source/algorithms/FindSXPeaksConvolve-v1.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+background shell). The integral of the kernel and background shell together is zero.
+
+The integrated intensity is evaluated by convolution of the signal array with the kernel and the square of the error on
+the integrated intensity is determined by convolution of the squared error array with the absolute value of the kernel.
+
+The threshold for peak detection is given by ``ThresholdIoverSigma`` which is the cutoff ratio of intensity/sigma (i.e.
+a valid peak would be expected to have intensity/sigma > 3). Stronger peaks will have a larger intensity/sigma.
+
+Useage
+-----------
```
[the remark this answers is not in the document]
Comment (RichardWaiteSTFC) [reply docs/source/algorithms/FindSXPeaksConvolve-v1.rst]: Good to know thanks! The OSX job was definitely run (recall previously problems with the new doc test failing!).

**Comment by sf1919** [comment]
@RichardWaiteSTFC As this has doc changes can you also run a `build_packages_from_branch` build but just for packaging Linux. If you don't know how to do this message me and I'll explain further

**Comment by RichardWaiteSTFC** [comment]
Thanks - have started the build package here
https://builds.mantidproject.org/job/build_packages_from_branch/554/

**Review by robertapplin — APPROVED** [review]
The code looks good, the tests all pass, functionally it also works as expected. The original build package from branch failed due to an unrelated error. I re-ran it on a different node and it passed fine https://builds.mantidproject.org/job/build_packages_from_branch/556/

Happy to approve this

**Pull request merged**